### PR TITLE
[IMP] l10n_be: revise CoA, terms and translations

### PR DESCRIPTION
**Pull request by dylankiss**
The Belgian localization has been updated to reflect the official Charts of Accounts.
Users now have the choice between the official CoA for Companies or the one for Associations and Foundations.

All related terms and translations have been reviewed and corrected in the 3 official languages (nl, fr, de).

Non-official languages have been removed, since they both don't make sense and they did not work because of targeting an ancient version of Odoo.

Finally, the 580000 account has been set as the default Internal Transfers account, avoiding a second 58 account to be created.

task-3086473

Related to https://github.com/odoo/enterprise/pull/35771
Related to https://github.com/odoo/upgrade/pull/5291

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_report.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ class AccountReportLine(models.Model):
     aggregation_formula = fields.Char(string="Aggregation Formula Shortcut", help="Internal field to shorten expression_ids creation for the aggregation engine", inverse='_inverse_aggregation_formula', store=False)
 
     _sql_constraints = [
-        ('code_uniq', 'unique (code)', "A report line with the same code already exists."),
+        ('code_uniq', 'unique (report_id, code)', "A report line with the same code already exists."),
```
Comment (Levizar) [inline addons/account/models/account_report.py:310]: @oco-odoo It would be nice to be able to specify the id of the report for cross report formula to avoid the issue mentioned in that commit.

> Currently we require the account lines `code` field to be unique across
all reports in a database. Since Reportalypse, this is not required
anymore, since we search by default only in the current report unless
we specify a `cross_report` subformula. Then it will take the first one
it finds.

File: addons/account/models/account_reconcile_model.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ class AccountReconcileModelLine(models.Model):
     # This field is ignored in a bank statement reconciliation.
     journal_id = fields.Many2one('account.journal', string='Journal', ondelete='cascade',
         domain="[('type', '=', 'general')]", check_company=True)
-    label = fields.Char(string='Journal Item Label')
+    label = fields.Char(string='Journal Item Label', translate=True)
```
Comment (Levizar) [inline addons/account/models/account_reconcile_model.py:59]: It seems unrelated to the task/commit purpose.
Why do we need to translate it now? Was it discussed/ask for by TSB?

File: addons/l10n_be/data/template/account.account-be.csv
Comment (Levizar) [inline addons/l10n_be/data/template/account.account-be.csv:1]: @tsb-odoo @qdp-odoo It's just rambling but I hate this `.csv` format for tax and account data.
It makes it nearly impossible to understand what changed at first glance. :sob:

File: addons/l10n_be/models/template_be_ass.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import _, models
+
+from odoo.addons.account.models.chart_template import template
+
+
+class AccountChartTemplate(models.AbstractModel):
+    _inherit = 'account.chart.template'
+
+    @template('be_ass')
```
Comment (Levizar) [inline addons/l10n_be/models/template_be_ass.py:10]: I think `be_ass` is a bit silly. It's ok for me to keep that "name" but do the others (mainly QDP and POs) agree with it?

**Review by Levizar — COMMENTED** [review]
GJ, this pr seems ok to me.
Just 2 dumb questions.

I wasn't really able to review the `account.account.csv` part as it makes the csv unreadable.
Do you think I should write a script to make some basic automated checks or your confident in your process writing them?

Edit: I still have to go through the enterprise PR

File: addons/l10n_be/models/template_be_ass.py (hunk 3 above)
Comment (tsb-odoo) [reply addons/l10n_be/models/template_be_ass.py:10]: May I suggest be_asso ?

**Comment by dylankiss** [comment]
> I wasn't really able to review the `account.account.csv` part as it makes the csv unreadable.
> Do you think I should write a script to make some basic automated checks or your confident in your process writing them?

@Levizar If you want to have a look at my process of dividing the accounts between the three files and the same for the reports, you can have a look at my working file in Sheets: https://docs.google.com/spreadsheets/d/1oohYs5Z6O0-JcU5e8jIJi5JpKuzkyaMfzvJ3tVfvP8w/edit?usp=sharing

I looked at the CBN/CNC PCMNs in all languages (they now have them in EN, FR, NL and DE, which is very convenient) and then of course the official BS/PL versions of the National Bank: https://www.nbb.be/fr/centrale-des-bilans/etablir-et-deposer/que-faut-il-deposer/modeles.

Feel free to go through it to check if your trained accountants' eye doesn't notice something weird. But I'm pretty confident I should have everything covered.

File: addons/l10n_be/data/template/account.account-be.csv
Comment (dylankiss) [reply addons/l10n_be/data/template/account.account-be.csv:1]: I actually quite liked working that way for a country with three official languages 😅

File: addons/l10n_be/models/template_be_ass.py (hunk 3 above)
Comment (dylankiss) [reply addons/l10n_be/models/template_be_ass.py:10]: It's part of the "funny messages" philosophy ... Come on! 😄 
No, for me it's fine changing it. So that could then be `be_asso` and `be_comp`?

File: addons/account/models/account_reconcile_model.py (hunk 2 above)
Comment (tsb-odoo) [reply addons/account/models/account_reconcile_model.py:59]: I didn't ask for it. Though, it sounds like a good initiative to me.

File: addons/account/models/account_report.py (hunk 1 above)
Comment (oco-odoo) [reply addons/account/models/account_report.py:315]: Maybe some day. Though that feature is not used that often, so there are probably other things to do first.

**Review by Levizar — APPROVED** [review]
LGTM for that PR
@robodoo r+

**Comment by Levizar** [comment]
@robodoo rebase-ff

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Comment by Levizar** [comment]
I just rebased the pr
@robodoo r+

**Comment by Levizar** [comment]
@robodoo r+
